### DISCUSSION TITLE
[Feature Request] Edge Scrolling

### DISCUSSION THREAD

**Discussion by ForsakenRei** [rustdesk/rustdesk#3053] [Ideas]
One thing I really missed is the edge scrolling from other alternatives.

Currently we have scroll bar and auto scrolling for the case you have a higher resolution screen on the remote side but a lower one as host. Auto scrolling is much better than manually use scroll bar when I have a veritcal screen remotely, while it will always keep scrolling as long as your cursor is moving and can be annoying some times when I want to see a fixed area but need move m cursor around it.

Edge scrolling here is, when you cursor hit one of the edges of rustdesk window, it will scroll to that direction, instead of keep moving when cursor moves within the window. For example, here's my rustdesk window,
![20230202221734 copy](https://user-images.githubusercontent.com/23041178/216507597-9640ee3f-1354-4f11-8427-ada22077b401.png)
when I move my cursor within the red edge(which is the display area of rustdesk window), even there are other stuff not displayed(e.g. title bar of Chrome) the display area won't move. If my cursor hit the red edge, which is the edge of the display area, content displayed will move accordingly based on which edge I hit. For example if my cursor hit the top edge then it will scroll up to show Chrome title bar.

Not sure how hard it will be to implement, and hope I was not the only one who think this is useful.

**Comment by logiclrd** [comment]
I too would much prefer this. :slightly_smiling_face: I find the auto scroll a bit disorienting. You aim for a pixel and by the time the mouse gets there, it's moved.

**Comment by logiclrd** [comment]
I have created PR #13247 that closes this issue. I forgot to tag it when initially creating the PR, so the mention that would normally appear here didn't happen, so I figured I'd add this comment by hand. :slightly_smiling_face:

**Reply by ForsakenRei** [reply]
Thank you.
